examples/sub_optim_prox.py

```diff
@@ -0,0 +1,146 @@
+        'to solve a TV regularized regression.')
+    parser.add_argument('--gpu', type=int, default=None,
+                        help='Use GPU <gpu> to run the computations. If it is '
+                        'not set, use CPU computations.')
+    parser.add_argument('--max-iter', type=int, default=300,
+                        help='Max number of iterations to train the '
+                        'learnable networks.')
+    parser.add_argument('--seed', type=int, default=None,
+                        help='Set the seed for the experiment. Can be used '
+                        'for debug or to freeze experiments.')
+    args = parser.parse_args()
+
+    if args.gpu is not None:
+        device = f"cuda:{args.gpu}"
+    else:
+        device = 'cpu'
+
+    print(__doc__)
+    print('*' * 80)
+
+    t0 = time.time()
+
+    ploting_dir = 'outputs_plots_sub_optimality_comparison'
+    if not os.path.exists(ploting_dir):
+        os.makedirs(ploting_dir)
+
+    ###########################################################################
+    # Define variables and data
+
+    # Define variables
+    n_samples = 100
+    n_samples_testing = 50
+    n_atoms = 8
+    n_dim = 5
+    s = 0.2
+    snr = 0.0
+    n_layers = 50
+    lbda = 0.5
+
+    seed = np.random.randint(0, 1000)
+    if args.seed is not None:
+        seed = args.seed
+    rng = check_random_state(seed)
+    print(f'Seed used = {seed}')  # noqa: E999
+
+    # Generate data
+    results = synthetic_1d_dataset(n_atoms=n_atoms, n_dim=n_dim, n=n_samples,
+                                   s=s, snr=snr, seed=seed)
+    x, _, z, L, D, A = results
+
+    x_train = x[n_samples_testing:, :]
+    x_test = x[:n_samples_testing, :]
+
+    ###########################################################################
+    # Main experiment
+    methods = [  # can be commented in one #
+        ('Analysis TV-Original', 'origtv',
+         dict(n_inner_layer=20), 'tab:orange', '', '--'),
+        ('Analysis TV-Original', 'origtv',
+         dict(n_inner_layer=200), 'tab:orange', '', 'solid'),
+        ('Analysis untrained-TV-Original', 'untrained-origtv',
+         dict(n_inner_layer=20), 'tab:blue', '', '--'),
+        ('Analysis untrained-TV-Original', 'untrained-origtv',
+         dict(n_inner_layer=200), 'tab:blue', '', 'solid'),
+    ]
+
+    memory = Memory('__cache_dir__', verbose=0)
+
+    @memory.cache
+    def run_experiment(methods, x_train, x_test, A, lbda, n_layers):
+        """ Experiment launcher. """
+        print("=" * 80)
+
+        l_diff_loss = []
+        for name, type_, kwargs, _, _, _ in methods:
+            print(f"[main script] running {name}")
+            print("-" * 80)
+
+            algo_type = 'origtv' if ('untrained' in type_) else type_
+            network = LearnTVAlgo(algo_type=algo_type, A=A, n_layers=n_layers,
+                                  max_iter=args.max_iter, device=device,
+                                  n_inner_layer=kwargs['n_inner_layer'],
```

Comment:
why don't use `**kwargs`?
this way your would be agnostic to the class and arguments